### DISCUSSION TITLE
"R for Windows" Upgrade doesn't work as it should

### DISCUSSION THREAD

**Discussion by Swe-it** [microsoft/winget-cli#2982] [General] [answered]
If you have an old version of R for Windows and run this:

`Winget Upgrade -e --id RProject.R`

It will install a newer version instead of updating, you then have two versions installed.

I don't know if this is by design, I just want to mention it if there is a configuration in the back that is not correct.

**Comment by mdanish-kh** [comment, marked as answer]
This is because the application installs "side-by-side" with previous versions. If the installer doesn't upgrade "in-place", we can add `UpgradeBehavior: uninstallPrevious` key in WinGet's manifest to tell WinGet to call the uninstaller first before installing the newer version. (There's also an experimental arg [--uninstall-previous](https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md#uninstallpreviousargument) to pass to WinGet install or upgrade commands to explicitly call the uninstaller before installing a package)

I've opened a PR to fix this issue:
- https://github.com/microsoft/winget-pkgs/pull/99054

For future package related issues, consider [opening an issue](https://github.com/microsoft/winget-pkgs/issues/new/choose) over at WinGet's Community Packages Repository.
